lib/guard/puma/runner.rb

```diff
@@ -43,11 +43,21 @@ def start
     end
 
     def halt
-      run_puma_command!("halt")
+      Net::HTTP.get build_uri('halt')
+
+      return true
+    rescue Errno::ECONNREFUSED
+      # server may not have been started correctly, but we are halting so who cares.
+      return true
```

Comment:
Let's try to keep the existing method and pass in the return value on error. It should probably default to `false`, or could take a block to get the `start` functionality and the `true/false`.